xds/src/main/java/io/grpc/xds/Bootstrapper.java

```diff
@@ -160,7 +160,8 @@ static BootstrapInfo parseConfig(String rawData) throws IOException {
     nodeBuilder.setUserAgentVersion(buildVersion.getImplementationVersion());
     nodeBuilder.addClientFeatures(CLIENT_FEATURE_DISABLE_OVERPROVISIONING);
 
-    return new BootstrapInfo(servers, nodeBuilder.build());
+    Map<String, ?> certProviders = JsonUtil.getObject(rawBootstrap, "certificate_providers");
```

Comment:
Do you really just want to keep the content of certificate_providers as a blob?

Reply:
So this one proposes the changes to bootstrap https://docs.google.com/document/d/1k9nnfZXCyuvWi3MsLSStPMGAyf-UEw2Mvso0ZxDTAG4/edit?ts=5f244687# - which says
>A new field will be added to the bootstrap file which will contain a map of CertificateProvider plugin instances (a plugin instance is a unique instantiation of a CertificateProvider with a specific configuration). The key to this map will be an opaque name (agreed upon by the client and the control plane) and the value will be the CertificateProvider name and configuration.

So the new field is "certificate_providers" and the value (JSON object) represents a map of Certificate-provider instance names to (plugin-name + configs) tuple. e.g.
```
{
    "gcp_id": {
        "plugin_name": "meshca",
        "config": { ...MeshCA config... }
    },
    "local_file": {
        "plugin_name": "file_watcher",
        "config": { ...FileWatcher config... }
    }
}
```
I will update this PR to represent this structure. Thanks

Reply:
Is there a definition for what a certificate provider plugin's config would look like? I saw that is a big blob. Hopefully we could have a Java object to represent that.

A parsed `CertificateProvider` will roughly look like:

```
class CertificateProvider {
  private final String name;
  private final String pluginName;
  private final Config config;  // To be defined.

  // Getters.
}

```

Reply:
> Is there a definition for what a certificate provider plugin's config would look like? I saw that is a big blob. Hopefully we could have a Java object to represent that.
> 
> A parsed `CertificateProvider` will roughly look like:
> 
> ```
> class CertificateProvider {
>   private final String name;
>   private final String pluginName;
>   private final Config config;  // To be defined.
> 
>   // Getters.
> }
> ```

No, take the `name` out and call it an instanceName. So one should be able to get the "pluginName" and "config" by looking up the instanceName. Also the Config is just a representation of JSON snippet so a `Map<String, ?>` should work.

Reply:
fixed as per the design described.

Reply:
> Also the Config is just a representation of JSON snippet so a Map<String, ?> should work.

A consumer of the certificate provider config would suffer from reading this `Map<String, ?>`. It needs to take steps as a parser to go inside the map, do casting and populate pieces out, which really should be done by parser.

Reply:
Agreed. But the current spec just dictates a JSON bob as the value of config. Also the CertificateProvider APIs are going to consume a generic Map as the JSON representation of a config. So converting this to a Java class is counter-productive.

```
"config" : { ...}
```